CONTRIBUTING.md

```diff
@@ -76,3 +76,128 @@ issues.
 Although not a formal channel, you can also find a number of core developers on
 the #php.pecl channel on [EFnet](http://www.efnet.org/). Similarly, many
 documentation writers can be found on #php.doc.
+
+## PHP source code directory structure
+
+PHP source code also includes several files generated during development and
+several parts where maintenance is happening upstream in their respective
+locations.
+
+```bash
+<php-src>/
+ ├─ .git/                           # Git configuration and source directory
+ └─ TSRM/                           # Thread Safe Resource Manager
+    └─ m4/                          # https://github.com/autoconf-archive/autoconf-archive
+       └─ ax_func_which_gethostbyname_r.m4
+    └─ ...
+ └─ Zend/                           # Zend Engine
+    ├─ zend_language_scanner.c      # Generated by re2c 1.0.3
```

Comment:
I'd drop the versions here. These will vary depending on who generated it last.

Reply:
Dropped all re2c versions except for the date/lib where 0.15.3 has been used so far. In case I should drop also these two, let me known. Thanks.